datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -3443,6 +3545,115 @@ mod tests {
+            ),
+        );
+        (schema, statistics)
+    }
+
+    #[test]
+    fn prune_utf8_eq() {
+        let (schema, statistics) = utf8_setup();
+
+        // Expression "s1 = 'A'"
+        // s1 ["A", "Z"] ==> some rows could pass (must keep)
+        // s1 ["A", "L"] ==> some rows could pass (must keep)
+        // s1 ["N", "Z"] ==> no rows can pass (not keep)
+        // s1 ["M", "M"] ==> no rows can pass (not keep)
+        // s1 [NULL, NULL]  ==> unknown (must keep)
+        // s1 ["A", NULL]  ==> unknown (must keep)
+        let expected_ret = &[true, true, false, false, true, true];
+
+        prune_with_expr(
+            // s1 LIKE 'A%'
+            col("s1").eq(lit("A")),
+            &schema,
+            &statistics,
+            expected_ret,
+        );
+    }
+
+    #[test]
+    fn prune_utf8_like() {
+        let (schema, statistics) = utf8_setup();
+
+        // Expression "s1 LIKE 'A%'"
+        // s1 ["A", "Z"] ==> some rows could pass (must keep)
+        // s1 ["A", "L"] ==> some rows could pass (must keep)
+        // s1 ["N", "Z"] ==> no rows can pass (not keep)
+        // s1 ["M", "M"] ==> no rows can pass (not keep)
+        // s1 [NULL, NULL]  ==> unknown (must keep)
+        // s1 ["A", NULL]  ==> unknown (must keep)
+        let expected_ret = &[true, true, false, false, true, true];
+
+        prune_with_expr(
+            // s1 LIKE 'A%'
+            col("s1").like(lit("A%")),
+            &schema,
+            &statistics,
+            expected_ret,
+        );
+    }
+
+    #[test]
+    fn prune_utf8_not_like() {
+        let (schema, statistics) = utf8_setup();
+
+        // Expression "s1 NOT LIKE 'M%'"
+        // s1 ["A", "Z"] ==> some rows could pass (must keep)
+        // s1 ["A", "L"] ==> some rows could pass (must keep)
+        // s1 ["N", "Z"] ==> some rows could pass (must keep)
+        // s1 ["M", "M"] ==> no rows can pass (not keep)
+        // s1 [NULL, NULL]  ==> unknown (must keep)
+        // s1 ["A", NULL]  ==> unknown (must keep)
+        let expected_ret = &[true, true, true, false, true, true];
+
+        prune_with_expr(
+            // s1 NOT LIKE 'M%'
+            col("s1").not_like(lit("M%")),
+            &schema,
+            &statistics,
+            expected_ret,
+        );
+    }
+
+    #[test]
+    fn prune_utf8_like_empty_suffix() {
+        let (schema, statistics) = utf8_setup();
+
+        // Expression "s1 LIKE '%A'"
+        // s1 ["A", "Z"] ==> some rows could pass (must keep)
+        // s1 ["A", "L"] ==> some rows could pass (must keep)
+        // s1 ["N", "Z"] ==> some rows could pass (must keep)
+        // s1 ["M", "M"] ==> some rows could pass (must keep)
+        // s1 [NULL, NULL]  ==> unknown (must keep)
+        // s1 ["A", NULL]  ==> unknown (must keep)
+        let expected_ret = &[true, true, true, true, true, true];
+
+        prune_with_expr(
+            // s1 LIKE 'A%'
```

Comment:
```suggestion
            // s1 LIKE '%A'
```

Reply:
this one is still wrong

Reply:
Yeah I haven't pushed pending discussing the correctness of the general approach.